drivers/hid/hidparse/parser.c

```diff
@@ -0,0 +1,935 @@
+/*
+ * PROJECT:     ReactOS HID Parser Library
+ * LICENSE:     GPL-3.0-or-later (https://spdx.org/licenses/GPL-3.0-or-later)
+ * PURPOSE:     HID Parser
+ * COPYRIGHT:   Copyright 2022 Roman Masanin <36927roma@gmail.com>
+ */
+
+/* Purpose of this file is to parse the HID report descriptor into preparsed data
```

Comment:
Why did you move this parser code from the universal library to KM library?

Reply:
HidP_GetCollectionDescription it is km only function

Reply:
actually here is more, because HidP_GetCollectionDescription also uses "PoolType" argument that must be used. I didn't realize this behavior for now, but it will be impossible in um/km mode.